go/gildedrose/Factory.go

```diff
@@ -0,0 +1,20 @@
+package gildedrose
+
+const (
+	brie     = "Aged Brie"
+	sulfuras = "Sulfuras, Hand of Ragnaros"
+	tafkal   = "Backstage passes to a TAFKAL80ETC concert"
+)
+
+func getUpdater(item *Item) (updater Updater) {
+	if item.Name == tafkal {
+		return NewTakfkalItem(item)
+	}
+	if item.Name == brie {
+		return NewBrieItem(item)
+	}
+	if item.Name == sulfuras {
+		return NewSulfItem(item)
+	}
+	return NewOtherItem(item)
+}
```

Comment:
Multiple return calls in a single function increase the cyclomatic complexity in your code. This design for this function will increase the cyclomatic complexity linearly.

https://en.wikipedia.org/wiki/Cyclomatic_complexity

You can refactor the cyclo complexity away by having 1 return at the end.

Then the problem becomes - what happens if the variable changes mid-flow? To alleviate that, use a switch call instead of many ifs. https://go.dev/tour/flowcontrol/9